flink-table/flink-table-common/src/main/java/org/apache/flink/table/types/inference/strategies/SpecificInputTypeStrategies.java

```diff
@@ -120,6 +124,41 @@ public final class SpecificInputTypeStrategies {
     public static final InputTypeStrategy TWO_EQUALS_COMPARABLE =
             comparable(ConstantArgumentCount.of(2), StructuredType.StructuredComparison.EQUALS);
 
+    /** Type strategy specific for {@link BuiltInFunctionDefinitions#MAP_KEYS}. */
+    public static final TypeStrategy MAP_KEYS =
+            callContext ->
+                    Optional.of(
+                            DataTypes.ARRAY(
+                                            callContext
+                                                    .getArgumentDataTypes()
+                                                    .get(0)
+                                                    .getChildren()
+                                                    .get(0))
+                                    .nullable());
+
```

Comment:
1. should use cast  KeyValueDataType and it will throw exception eagerly
 Optional.of(
                                                            DataTypes.ARRAY(
                                                                    ((KeyValueDataType) mapType)
                                                                            .getKeyDataType()))

2. why force nullable, it may not be right

Reply:
Makes sense, thanks
done